presto-thrift-connector/src/test/java/com/facebook/presto/connector/thrift/integration/TestThriftIntegrationSmokeTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.thrift.integration;
+
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.tests.AbstractTestIntegrationSmokeTest;
+import com.facebook.swift.service.ThriftServer;
+import com.google.common.io.Closeables;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.facebook.presto.connector.thrift.integration.ThriftQueryRunner.createThriftQueryRunner;
+import static com.facebook.presto.connector.thrift.integration.ThriftQueryRunner.startThriftServers;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.tests.QueryAssertions.assertContains;
+import static java.util.Objects.requireNonNull;
+
+public class TestThriftIntegrationSmokeTest
+        extends AbstractTestIntegrationSmokeTest
+{
+    private final List<ThriftServer> thriftServers;
+
+    public TestThriftIntegrationSmokeTest()
+            throws Exception
+    {
+        this(startThriftServers(2));
+    }
+
+    public TestThriftIntegrationSmokeTest(List<ThriftServer> servers)
+            throws Exception
+    {
+        super(() -> createThriftQueryRunner(servers, 2));
```

Comment:
I re-worked test setup part. Now it closes thrift servers during query runner close and catches exceptions during setup.